eZ/Bundle/EzPublishIOBundle/Resources/config/io.yml

```diff
@@ -1,8 +1,92 @@
+    ezpublish.core.io.migration.file_lister_registry.class: eZ\Bundle\EzPublishIOBundle\Migration\FileListerRegistry\IdentifierBased
+    ezpublish.core.io.migration.file_lister.binary_file_lister.class: eZ\Bundle\EzPublishIOBundle\Migration\FileLister\BinaryFileLister
+    ezpublish.core.io.migration.file_lister.image_file_lister.class: eZ\Bundle\EzPublishIOBundle\Migration\FileLister\ImageFileLister
+    ezpublish.core.io.migration.file_lister.media_file_lister.class: eZ\Bundle\EzPublishIOBundle\Migration\FileLister\BinaryFileLister
+    ezpublish.core.io.migration.file_lister.file_iterator.binary_file_iterator.class: eZ\Bundle\EzPublishIOBundle\Migration\FileLister\FileIterator\LegacyStorageFileIterator
+    ezpublish.core.io.migration.file_lister.file_iterator.media_file_iterator.class: eZ\Bundle\EzPublishIOBundle\Migration\FileLister\FileIterator\LegacyStorageFileIterator
+    ezpublish.core.io.migration.file_lister.file_row_reader.binary_file_row_reader.class: eZ\Bundle\EzPublishIOBundle\Migration\FileLister\FileRowReader\LegacyStorageBinaryFileRowReader
+    ezpublish.core.io.migration.file_lister.file_row_reader.media_file_row_reader.class: eZ\Bundle\EzPublishIOBundle\Migration\FileLister\FileRowReader\LegacyStorageMediaFileRowReader
+    ezpublish.core.io.migration.file_migrator.class: eZ\Bundle\EzPublishIOBundle\Migration\FileMigrator\FileMigrator
+    ezpublish.core.io.migration.migration_handler.class: eZ\Bundle\EzPublishIOBundle\Migration\MigrationHandler
     ezpublish.core.io.stream_file_listener.class: eZ\Bundle\EzPublishIOBundle\EventListener\StreamFileListener
     ezpublish.core.io.flysystem.default_adapter.class: League\Flysystem\Adapter\Local
 
 services:
+    ezpublish.core.io.command.migrate_files:
+        class: "%ezpublish.core.io.command.migrate_files.class%"
+        arguments:
+            - "%ez_io.metadata_handlers%"
+            - "%ez_io.binarydata_handlers%"
+            - "@ezpublish.core.io.migration.file_lister_registry"
+            - "@ezpublish.core.io.migration.file_migrator"
+        tags:
+            - { name: console.command }
+
+    ezpublish.core.io.migration.file_lister_registry:
+        class: "%ezpublish.core.io.migration.file_lister_registry.class%"
+
+    ezpublish.core.io.migration.migration_handler:
+        class: "%ezpublish.core.io.migration.migration_handler.class%"
+        arguments:
+            - "@ezpublish.core.io.metadata_handler.factory"
+            - "@ezpublish.core.io.binarydata_handler.factory"
+            - "@logger"
+
+    ezpublish.core.io.migration.file_lister.binary_file_lister:
+        class: "%ezpublish.core.io.migration.file_lister.binary_file_lister.class%"
+        parent: ezpublish.core.io.migration.migration_handler
+        arguments:
+            - "@ezpublish.core.io.migration.file_lister.file_iterator.binary_file_iterator"
+            - "%ezsettings.default.binary_dir%"
```

Comment:
@bdunogier Yes I'd like to get the setting from the current siteaccess, but `$binary_dir$` is apparently not it, it gets interpreted as a literal string.